### PR TITLE
TPRUN-8070: Feature dependency camel-google-storage/0.0.0 is not available

### PR DESCRIPTION
**Pull request by stataru8**
🏁 **Context**
- [TPRUN-8070](https://jira.talendforge.org/browse/TPRUN-8070)

🔍 **What is the problem this PR is trying to solve?**
camel-google-storage component is not currently supported in Camel-Karaf

🚀 **What is the chosen solution to this problem?**
Add camel-google-storage feature to camel karaf features file

🎾 **Impacts**

- [ ] **This PR introduces a breaking change**

**Dear contributor, please check if the PR fulfills these requirements**

- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

### CODE REVIEW

File: platforms/karaf/features/src/main/resources/features.xml [hunk 1]
```diff
@@ -1069,6 +1069,35 @@
...
+    <bundle dependency='true'>wrap:mvn:com.google.oauth-client/google-oauth-client/${google-cloud-oauth-client-version}$Bundle-SymbolicName=com.google.oauth-client.google-oauth-client&amp;Bundle-Version=${google-cloud-oauth-client-version}$overwrite=merge&amp;Import-Package=com.google.common.base;version="${google-cloud-guava.tesb.version-range}",*</bundle>
+    <bundle dependency='true'>wrap:mvn:io.opencensus/opencensus-api/${google-http-client-opencensus-api-version}$Bundle-SymbolicName=io.opencensus.opencensus-api&amp;Bundle-Version=${google-http-client-opencensus-api-version}</bundle>
+    <bundle dependency='true'>wrap:mvn:io.opencensus/opencensus-contrib-http-util/${google-http-client-opencensus-api-version}$Bundle-SymbolicName=io.opencensus.opencensus-contrib-http-util&amp;Bundle-Version=${google-http-client-opencensus-api-version}</bundle>
+    <bundle>wrap:mvn:com.google.api-client/google-api-client/${google-cloud-api-client-version}$Bundle-SymbolicName=com.google.api-client.google-api-client&amp;Bundle-Version=${google-cloud-api-client-version}$overwrite=merge&amp;Import-Package=com.google.common.base;version="${google-cloud-guava.tesb.version-range}",*</bundle>
```
Comment (stataru8) [inline platforms/karaf/features/src/main/resources/features.xml:1098]: I don't know why but, setting `dependency='true'` for this bundle prevents it from being installed whenever the feature is installed. You can do a quick test by executing:
`feature:info camel-google-bigquery` (check that this already existing feature requires this bundle)
`feature:list | grep camel-google-bigquery` (verify that the feature isn't installed))
`feature:install camel-google-bigquery` 
`feature:list | grep camel-google-bigquery` (verify that the feature is installed))
`list -t 0 -u | grep google-api-client` (verify that this bundle wasn't installed)

File: platforms/karaf/features/src/main/resources/features.xml [hunk 2]
```diff
@@ -1069,6 +1069,35 @@
...
+    <bundle dependency='true'>wrap:mvn:com.google.cloud/google-cloud-storage/${google-cloud-storage.tesb.version}$Bundle-SymbolicName=com.google.cloud.google-cloud-storage&amp;Bundle-Version=${google-cloud-storage.tesb.version}</bundle>
+    <bundle dependency='true'>wrap:mvn:com.google.http-client/google-http-client-appengine/${google-cloud-http-client-version}$Bundle-SymbolicName=com.google.http-client.google-http-client-appengine&amp;Bundle-Version=${google-cloud-http-client-version}</bundle>
+    <bundle dependency='true'>wrap:mvn:com.google.http-client/google-http-client-jackson2/${google-cloud-http-client-version}$Bundle-SymbolicName=com.google.http-client.google-http-client-jackson2&amp;Bundle-Version=${google-cloud-http-client-version}</bundle>
+    <bundle dependency='true'>wrap:mvn:io.grpc/grpc-context/${grpc-context.tesb.version}$Bundle-SymbolicName=io.grpc.grpc-context&amp;Bundle-Version=${grpc-context.tesb.version}</bundle>
+    <!-- google-api-client -->
```
Comment (stataru8) [inline platforms/karaf/features/src/main/resources/features.xml:1087]: According to the verify plugin, the `google-api-client` bundle requires every bundle listed below. I have chosen not to create a feature for it, but we could....

File: platforms/karaf/features/src/main/resources/features.xml [hunk 3]
```diff
@@ -1069,6 +1069,35 @@
...
+    <bundle dependency='true'>wrap:mvn:com.google.oauth-client/google-oauth-client/${google-cloud-oauth-client-version}$Bundle-SymbolicName=com.google.oauth-client.google-oauth-client&amp;Bundle-Version=${google-cloud-oauth-client-version}&amp;overwrite=merge&amp;Import-Package=com.google.common.base;version="${google-cloud-guava.tesb.version-range}",*</bundle>
+    <bundle dependency='true'>wrap:mvn:io.opencensus/opencensus-api/${google-http-client-opencensus-api-version}$Bundle-SymbolicName=io.opencensus.opencensus-api&amp;Bundle-Version=${google-http-client-opencensus-api-version}</bundle>
+    <bundle dependency='true'>wrap:mvn:io.opencensus/opencensus-contrib-http-util/${google-http-client-opencensus-api-version}$Bundle-SymbolicName=io.opencensus.opencensus-contrib-http-util&amp;Bundle-Version=${google-http-client-opencensus-api-version}</bundle>
+    <bundle>wrap:mvn:com.google.api-client/google-api-client/${google-cloud-api-client-version}$Bundle-SymbolicName=com.google.api-client.google-api-client&amp;Bundle-Version=${google-cloud-api-client-version}&amp;overwrite=merge&amp;Import-Package=com.google.common.base;version="${google-cloud-guava.tesb.version-range}",*</bundle>
```
Comment (stataru8) [inline platforms/karaf/features/src/main/resources/features.xml:1098]: I don't know why but, setting dependency='true' for this bundle prevents it from being installed whenever the feature is installed. You can do a quick test by executing:
feature:info camel-google-bigquery | grep api-client (check that this already existing feature requires this bundle with dependency='true')
feature:list | grep camel-google-bigquery (verify that the feature isn't installed))
feature:install camel-google-bigquery
feature:list | grep camel-google-bigquery (verify that the feature is installed))
list -t 0 -u | grep google-api-client (verify that this bundle wasn't installed)

File: platforms/karaf/features/src/main/resources/features.xml [hunk 4]
```diff
@@ -1069,6 +1069,35 @@
...
+    <bundle dependency='true'>wrap:mvn:com.google.oauth-client/google-oauth-client/${google-cloud-oauth-client-version}$Bundle-SymbolicName=com.google.oauth-client.google-oauth-client&amp;Bundle-Version=${google-cloud-oauth-client-version}&amp;overwrite=merge&amp;Import-Package=com.google.common.*;version="${google-cloud-guava.tesb.version-range}",*</bundle>
+    <bundle dependency='true'>wrap:mvn:io.opencensus/opencensus-api/${google-http-client-opencensus-api-version}$Bundle-SymbolicName=io.opencensus.opencensus-api&amp;Bundle-Version=${google-http-client-opencensus-api-version}</bundle>
+    <bundle dependency='true'>wrap:mvn:io.opencensus/opencensus-contrib-http-util/${google-http-client-opencensus-api-version}$Bundle-SymbolicName=io.opencensus.opencensus-contrib-http-util&amp;Bundle-Version=${google-http-client-opencensus-api-version}</bundle>
+    <bundle>wrap:mvn:com.google.api-client/google-api-client/${google-cloud-api-client-version}$Bundle-SymbolicName=com.google.api-client.google-api-client&amp;Bundle-Version=${google-cloud-api-client-version}&amp;overwrite=merge&amp;Import-Package=com.google.common.*;version="${google-cloud-guava.tesb.version-range}",*</bundle>
```
Comment (stataru8) [inline platforms/karaf/features/src/main/resources/features.xml:1098]: I don't know why but, setting `dependency='true'` for this bundle prevents it from being installed whenever the feature is installed. You can do a quick test by executing:
`feature:info camel-google-bigquery | grep api-client` (check that this already existing feature requires this bundle with `dependency='true'`)
`feature:list | grep camel-google-bigquery` (verify that the feature isn't installed))
`feature:install camel-google-bigquery` 
`feature:list | grep camel-google-bigquery` (verify that the feature is installed))
`list -t 0 -u | grep google-api-client` (verify that this bundle wasn't installed)

File: platforms/karaf/features/src/main/resources/features.xml (hunk 2 above)
Comment (andreasmattes) [reply platforms/karaf/features/src/main/resources/features.xml:1087]: Let's leave it this way for now, we may later define sets of Google jars as dependencies for Camel Google support.

File: platforms/karaf/features/src/main/resources/features.xml (hunk 4 above)
Comment (andreasmattes) [reply platforms/karaf/features/src/main/resources/features.xml:1098]: The evaluation of "dependency='true'" may not work properly with all the wrapped JARs. It is supposedly not installed as dependency if all imports have been calculated as optional or get resolved from elsewhere.

Comment (stataru8) [reply platforms/karaf/features/src/main/resources/features.xml:1098]: `bundle:tree-show <bundle_id>`

**Pull request merged**